### PR TITLE
feat: automatic proposal querying

### PR DESCRIPTION
**Pull request by maxwellmattryan**
## Summary

> Please summarize your changes, describing __what__ they are and __why__ they were made.

Official proposals should be automatically registered to the user's Governance dashboard.

## Changelog

```
- Add registerProposalsFromNode function called in login
- Add registerProposalsFromPrimaryNode
- Cleanup imports in login file
- Register proposals when new primary node is selected
- Set primary node during onboarding
```

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
	- [ ] MacOS
	- [x] Linux
	- [ ] Windows
- __Mobile__
	- [ ] iOS
	- [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

Create new profiles connected to node with voting events. Log into the profile more than once.

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by maxwellmattryan** [comment]
@Tuditi One change I think can be made is that we don't register a proposal if it has already ended, but some things worth noting:
- Some users may want to be able to see all proposals, even if they've ended, so that they can see the results of the vote
- Checking the status means adding an extra API call per proposal ID that we get. We could suggest for the node team to change this API endpoint so that it returns the entire event rather just the ID, which would be sensible since it's unlikely that there will be many many events on a node anyway (but who knows?)

My thoughts are that we should just keep it simple for now and register everything, then we can see based on beta testing how users feel about having every single event registered.

File: packages/shared/lib/contexts/governance/utils/getVotingEventIds.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { get } from 'svelte/store'
+
+import { ParticipationEventId, ParticipationEventType } from '@iota/wallet/out/types'
+
+import { selectedAccount } from '@core/account/stores'
+
+export function getVotingEventIds(): Promise<ParticipationEventId[]> {
```
Comment (nicole-obrien) [inline packages/shared/lib/contexts/governance/utils/getVotingEventIds.ts:7]:
```suggestion
export function getVotingParticipationEventIds(): Promise<ParticipationEventId[]> {
```

File: packages/shared/lib/contexts/governance/utils/index.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ export * from './getNumberOfVotedProposals'
 export * from './getNumberOfVotingProposals'
 export * from './getParticipationsForProposal'
 export * from './getPercentagesFromAnswerStatuses'
+export * from './getVotingEventIds'
```
Comment (nicole-obrien) [inline packages/shared/lib/contexts/governance/utils/index.ts:11]:
```suggestion
export * from './getVotingParticipationEventIds'
```

File: packages/shared/lib/contexts/governance/actions/registerProposalsFromPrimaryNode.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { get } from 'svelte/store'
+
+import { selectedAccount } from '@core/account/stores'
+import { activeProfile } from '@core/profile/stores'
+
+import { getVotingEventIds, isProposalAlreadyAddedForSelectedAccount } from '../utils'
+
+import { registerParticipationEvent } from './registerParticipationEvent'
+
+export async function registerProposalsFromPrimaryNode(): Promise<void> {
```
Comment (nicole-obrien) [inline packages/shared/lib/contexts/governance/actions/registerProposalsFromPrimaryNode.ts:10]: Can this be more generic by just passing in a node and then we can decide what nodes to use where we call this. i.e. we don't usually have a primary node

File: packages/shared/lib/contexts/governance/actions/registerProposalsFromPrimaryNode.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    if (!proposalIds || proposalIds.length === 0) {
+        return
+    }
+
+    const clientOptions = get(activeProfile)?.clientOptions
+    if (!clientOptions) {
+        throw new Error('Unable to retrieve profile client options')
+    }
+
+    const primaryNode = clientOptions?.primaryNode ?? clientOptions?.nodes[0]
```
Comment (nicole-obrien) [inline packages/shared/lib/contexts/governance/actions/registerProposalsFromPrimaryNode.ts:21]: We should just pass in the nodes, and we can call this in a wrapper function for all the nodes the user has.

**Review by nicole-obrien — CHANGES_REQUESTED** [review]
Left some comments, I recently cleaned up all our api requests so its not taking as long to load so we should be able to pull in all events from our nodes at login. 

We could also register the events for a node when we add it.

I also need to double check but the questions where not showing on the proposal details page 🤷‍♀️

**Comment by nicole-obrien** [comment]
> Checking the status means adding an extra API call per proposal ID that we get. We could suggest for the node team to change this API endpoint so that it returns the entire event rather just the ID, which would be sensible since it's unlikely that there will be many many events on a node anyway (but who knows?)

I have already changed this, we a don't need any api call check the status of an event it can be calculated by the current milestone.

File: packages/shared/lib/contexts/governance/actions/registerProposalsFromPrimaryNode.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { get } from 'svelte/store'
+
+import { activeProfile } from '@core/profile/stores'
+
+import { registerProposalsFromNode } from './registerProposalsFromNode'
+
+export async function registerProposalsFromPrimaryNode(): Promise<void> {
```
Comment (nicole-obrien) [inline packages/shared/lib/contexts/governance/actions/registerProposalsFromPrimaryNode.ts:7]: If we don't have a primary node we do not need to throw an error, just dont attempt to register any proposals for that node.

File: packages/shared/lib/contexts/governance/actions/registerProposalsFromNode.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import { INode } from '@core/network/interfaces'
+
+import { getVotingParticipationEventIds, isProposalAlreadyAddedForSelectedAccount } from '../utils'
+
+import { registerParticipationEvent } from './registerParticipationEvent'
+
+export async function registerProposalsFromNode(node: INode): Promise<void> {
+    if (!node) {
+        throw new Error('Unable to determine node')
+    }
```
Comment (Tuditi) [inline packages/shared/lib/contexts/governance/actions/registerProposalsFromNode.ts:13]: This error is redundant if you follow Nicole's suggestion.

**Review by Tuditi — APPROVED** [review]
Looks good, I left a comment wrt Nicole's requested changes.
Additionally, we'll need a way to store removed event IDs, since removed official events get added back on login.

**Review by Tuditi — CHANGES_REQUESTED** [review]
Tests are failing

**Comment by maxwellmattryan** [comment]
@Tuditi

**Comment by maxwellmattryan** [comment]
@Tuditi,

> Additionally, we'll need a way to store removed event IDs, since removed official events get added back on login.

Agreed, but per the call we had today let's do this as a separate task so we can get a release out.

**Review by nicole-obrien — APPROVED** [review]
LGTM!

**Pull request merged**